clj-http-lite/src/martian/clj_http_lite.clj

```diff
@@ -23,13 +24,17 @@
 (defn bootstrap [api-root concise-handlers & [opts]]
   (martian/bootstrap api-root concise-handlers (merge default-opts opts)))
 
-(defn bootstrap-openapi [url & [opts get-swagger-opts]]
+(defn bootstrap-openapi [url & [{:keys [server-url] :as opts} get-swagger-opts]]
   (let [definition (-> (http/get url (or get-swagger-opts {})) :body (json/parse-string keyword)) ;; clj-http-lite does not support {:as :json} body conversion (yet) so we do it right here
         {:keys [scheme server-name server-port]} (http/parse-url url)
-        base-url (format "%s://%s%s%s" (name scheme) server-name (if server-port (str ":" server-port) "")
-                         (if (openapi-schema? definition)
-                           (openapi/base-url definition)
-                           (get definition :basePath "")))]
+        api-root (or server-url (openapi/base-url definition))
```

Comment:
I think there is enough repeated code here across the modules to be extracted as a function somewhere in core. It's important that they stay consistent and with the number of modules it's starting to get unwieldy

Reply:
Fair enough.

I'll do that.

Reply:
Is `openapi/base-url` considered to be a part of the public API?
If not, are you okay if I change `openapi/base-url` to encapsulate the common code?
The return value seems to be used only in the `<module>.clj[s]` files for the `openapi-bootstrap` function.